### DISCUSSION TITLE
Emitting CREATE SEQUENCE statements when creating tables via CreateTable

### DISCUSSION THREAD

**Discussion by tonycosentini** [sqlalchemy/sqlalchemy#12000] [Usage Questions] [answered]
Hi,

I'm trying to print the DDL for creating a table but I can't figure out how to print out the statements to create sequences.

Here's a minimal example:
```python3
from sqlalchemy import Table, MetaData, Column, Integer, Sequence
from sqlalchemy.schema import CreateTable

from sqlalchemy.dialects import postgresql

metadata = MetaData()

some_table = Table(
    'some_table', metadata,
    Column('id', Integer, Sequence('some_table_seq', start=1),
    primary_key=True)
)

print(CreateTable(some_table).compile(dialect=postgresql.dialect()).string)
```

This prints:
```sql
CREATE TABLE some_table (
	id INTEGER NOT NULL, 
	PRIMARY KEY (id)
)
```

The docs say:
> When CREATE TABLE is emitted for the above [Table](https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.Table), if the target platform supports sequences, a CREATE SEQUENCE statement will be emitted as well. For platforms that don’t support sequences, the [Sequence](https://docs.sqlalchemy.org/en/20/core/defaults.html#sqlalchemy.schema.Sequence) construct is ignored.

However my hunch is this is only the case when creating the entire schema.

Is there any way to get the sequences emitted for a specific table? Alternatively I can probably use `CreateSequence`, but I couldn't find a way to find the sequences defined on the columns.

I'm testing with SQLAlchemy 2.0.35. As always thanks for reading!

**Comment by CaselIT** [comment]
Hi,

The sequence should be in the server_default of the column. Note that server default can have other objects as well, so you should check that it's a Sequence.

Note that you can use identity columns instead of sequences, using https://docs.sqlalchemy.org/en/20/core/defaults.html#sqlalchemy.schema.Identity

**Reply by tonycosentini** [reply]
This seems to imply that it can be inlined and `CreateTable` will include it - https://docs.sqlalchemy.org/en/20/core/defaults.html#sqlalchemy.schema.Sequence

That's where I got the example from.

**Reply by CaselIT** [reply, marked as answer]
As far as I know you can't inline a create sequence inside a create table, so no, that's not what it does.
That line means that when you execute `create()` on a table the sequence will be created as well, but in a separate create statement.

**Reply by tonycosentini** [reply]
Ahhh I see, that makes more sense. I will create a sequence outside of the table and call `CreateSequence` on that. Thanks for clarifying!
